### PR TITLE
functionality to add votes on articles & updates the database via api

### PR DESCRIPTION
**Pull request by tashkenna**

### CODE REVIEW

**Review by robert112233 — CHANGES_REQUESTED** [review]
Hi great work! Good use of optimistic rendering. I think you could consider the sad path more from a UX perspective. For example what happens if I accidentally vote on an article? My vote is forever cast and theres no way to undo it 😭 Maybe instead of disabling the button you decrement it on a second click? and toggle between incrementing and decrementing. 

I would also consider some sort of error being rendering if your fetch request doesn't go through. Currently when I vote offline the vote simply increases and I think everything's gone ahead correctly. You could decrement the vote back down in a catch block and change an isError state to true, rendering a message to say "somethings gone wrong, please try again" etc.

Sorry for the essay! Please submit another PR for review :)

File: src/components-pages/Article.jsx [hunk 1]
```diff
@@ -7,20 +7,39 @@ export const Article = () => {
...
+  const [isError, setIsError] = useState(false)
 
   useEffect(() => {
     getArticleByID(id).then(({ article }) => {
       setArticle(article);
       setLoading(false);
-    });
+    })
+    .catch((err) => {
+      console.log(err)
```
Comment (HeatherEver) [inline src/components-pages/Article.jsx]: make sure console logs are removed before pushing :)

File: src/components-pages/Article.jsx [hunk 2]
```diff
@@ -7,20 +7,39 @@ export const Article = () => {
...
+  const handleClick = () => {
+    const newVotes = userVotes > 0 ? userVotes - 1 : userVotes + 1;
 
+    patchArticleVotes(id, newVotes)
+      .then(() => {
+        setUserVotes(newVotes);
+        setIsError(false)
+      })
+      .catch((err) => {
+        console.log(err)
```
Comment (HeatherEver) [inline src/components-pages/Article.jsx]: and another

**Review by HeatherEver — COMMENTED** [review]
This is looking better, especially for if a user's vote does not go through! ⭐  However, if i vote on an article, then remove it - it works until i refresh the page and then the vote is back (please see my terrible screen recording for an example) 

https://github.com/tashkenna/fe-nc-news/assets/79215884/fa0dfbc9-33ef-49ae-a746-6e5ae87aec9c

**Review by yalicee — APPROVED** [review]
Thank you for making the changes, looks good :)

**Pull request merged**